source/write/replace.txt

```diff
@@ -0,0 +1,202 @@
+value is ``"Primola Restaurant"``:
+
+.. literalinclude:: /includes/write/replace.kt
+   :start-after: start-replace-one
+   :end-before: end-replace-one
+   :language: kotlin
+   :copyable:
+   :dedent:
+
+.. important::
+
+   The values of ``_id`` fields are immutable. If your replacement document specifies
+   a value for the ``_id`` field, it must be the same as the ``_id`` value of the
+   existing document.
+
+Customize the Replace Operation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``replaceOne()`` method optionally accepts
+a parameter that sets options to configure the replace operation.
+If you don't specify any options, the driver performs the replace
+operation with default settings.
+
+The following table describes the setter methods that you can use to
+configure an ``ReplaceOptions`` instance:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Property
+     - Description
+
+   * - ``upsert()``
+     - | Specifies whether the replace operation performs an upsert operation if no 
+         documents match the query filter. For more information, see :manual:`upsert
+         behavior </reference/method/db.collection.replaceOne/#upsert>`
+         in the {+mdb-server+} manual.
+       | Defaults to ``false``
+
+   * - ``bypassDocumentValidation()``
+     - | Specifies whether the update operation bypasses document validation. This lets you 
+         update documents that don't meet the schema validation requirements, if any 
+         exist. For more information about schema validation, see :manual:`Schema
+         Validation </core/schema-validation/#schema-validation>` in the MongoDB
+         Server manual.
+       | Defaults to ``false``.
+
+   * - ``collation()``
+     - | Specifies the kind of language collation to use when sorting
+         results. For more information, see :manual:`Collation </reference/collation/#std-label-collation>`
+         in the {+mdb-server+} manual.
+
+   * - ``hint()``
+     - | Sets the index to use when matching documents. 
+         For more information, see the :manual:`hint statement
+         </reference/method/db.collection.replaceOne/#std-label-replace-one-hint>`
+         in the {+mdb-server+} manual.
+
+   * - ``let()``
+     - | Provides a map of parameter names and values to set top-level
+         variables for the operation. Values must be constant or closed
+         expressions that don't reference document fields.
+
+   * - ``comment()``
+     - | Sets a comment to attach to the operation.
+
+The following code uses the ``replaceOne()`` method to match the first
+document in which the value of the ``name`` field is ``"Pizza Plus"``,
+then replaces the fields with new fields.
+
+Because the ``upsert`` option is set to ``true``, the driver inserts a
+new document that has the fields and values specified in the replacement
+document if the query filter doesn't match any existing documents.
+
+.. literalinclude:: /includes/write/replace.kt
```

Comment:
This code doesn't seem to show that the value of the name field is Pizza Plus

This is the code I see:
val opts = ReplaceOptions().upsert(true)
val result = collection.replaceOne(filter, replacement, opts)

Reply:
I'll edit the description, i dont think its necessary to show the filter